iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -98,6 +105,9 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             {
                 using var ctb = new CancellationTokenBundle(_operationTimeout, cancellationToken);
                 await _amqpUnit.OpenAsync(ctb.Token).ConfigureAwait(false);
+
+                // Create a timer to remove any expired messages. The timer would invoke callback after every hour.
+                _twinTimeoutTimer = new Timer(CheckTimeout, null, s_twinResponseTimeout, s_twinResponseTimeout);
```

Comment:
I feel like we should tie the lifetime to open/close instead, and double check disposal in Dispose.

That way it doesn't fire when we're not connected. Thoughts?

Reply:
Agree, didn't consider it from this point of view earlier.
Added change.